woocommerce/woocommerce-android
# [Order Creation] Handle variations sorting

Closes: #5725 


### Description
This PR makes `ProductVariation` a `Comparable` class, to allow sorting the list of variations using the `menu_order` property.
When two variations have the same value for `menu_order`, they will be sorted using their name generated from the attributes list.

The PR has an additional change in the commit 0ed0800, which removes the class `Option`, and keeps only `VariationOption`, as I think they represent the same thing, please review it, and confirm if my thinking is correct.

### Testing instructions
Change sorting of variations on wp-admin, and confirm the order is then respected in the app in both product details and order creation screens.
